builder.md

```diff
@@ -1,57 +1,180 @@
-
-In the case where schemes share the same slug, a builder will overwrite files previously generated from the template. Should this happen, a builder should show warning messages listing the overwritten files.
-
-## Template Variables
-A builder should provide the following variables to a template file:
+A builder tool MUST provide the following variables, and no others, to the
+template files it processes:
 
 - `scheme-name` - obtained from the scheme file
 - `scheme-author` - obtained from the scheme file
 - `scheme-slug` - obtained from the scheme filename, as described above
+- `scheme-kind` - either "light" or "dark", calculated from the `base00` color <!-- TODO: This is a candidate for inclusion, let me know your thoughts -->
+
+- `base00-hex` to `base0F-hex` - obtained from the scheme file e.g "7cafc2"
 - `base00-hex-r` to `base0F-hex-r` - built from the hex value in the scheme file e.g "7c"
 - `base00-hex-g` to `base0F-hex-g` - built from the hex value in the scheme file e.g "af"
 - `base00-hex-b` to `base0F-hex-b` - built from the hex value in the scheme file e.g "c2"
+- `base00-hex-bgr` to `base0F-hex-bgr` - built from a reversed version of all the hex values e.g "c2af7c"
+
 - `base00-rgb-r` to `base0F-rgb-r` - converted from the hex value in the scheme file e.g "124"
 - `base00-rgb-g` to `base0F-rgb-g` - converted from the hex value in the scheme file e.g "175"
 - `base00-rgb-b` to `base0F-rgb-b` - converted from the hex value in the scheme file e.g "194"
 - `base00-dec-r` to `base0F-dec-r` - converted from the rgb value in the scheme file e.g "0.87..."
 - `base00-dec-g` to `base0F-dec-g` - converted from the rgb value in the scheme file e.g "0.50..."
 - `base00-dec-b` to `base0F-dec-b` - converted from the rgb value in the scheme file e.g "0.21..."
 
-Builders should also provide the following variables for convenience:
+## 3. Interface
 
-- `base00-hex` to `base0F-hex` - obtained from the scheme file e.g "7cafc2"
-- `base00-hex-bgr` to `base0F-hex-bgr` - built from a reversed version of all the hex values e.g "c2af7c"
+All base16 reference builders MUST provide a single feature: building a
+template using 1 or more schemes.
+
+This section (`3`) is not relevant to extended tooling, which will have its own
+interface fitting their intended usecases, altough they MAY expose the
+reference CLI interface (through a `build` subcommand, for example) and/or a
+library, if desired.
+
+It is REQUIRED that this functionality is exposed by one (or both) of the following:
+
+### 3.1. CLI
+
+The binary name SHOULD contain `base16`, but is otherwise left as a choice to
```

Comment:
I'd argue that the binary name should contain `base16-builder`, especially for lower level plumbing.